agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -40,6 +65,7 @@ def apply
           warn "failed to sync #{service_pod.name} at #{service_pod.deploy_rev}: #{error}"
           sync_state_to_master(current_state, error)
         else
+          @container_state_changed = false
```

Comment:
This also kinda breaks without #2138 to raise on container start errors:

`kontena service create -v /dev/null/wtf:/dev/wtf test-fail redis`
`kontena service deploy test-fail`

```
D, [2017-04-12T07:34:56.511908 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state of test-fail-1: running
I, [2017-04-12T07:34:56.514946 #1]  INFO -- Kontena::Workers::ServicePodWorker: creating test-fail-1
I, [2017-04-12T07:34:56.515191 #1]  INFO -- Kontena::ServicePods::Creator: creating service: test-fail-1

D, [2017-04-12T07:34:59.044802 #1] DEBUG -- Kontena::ServicePods::Creator: creating container: test-fail-1
D, [2017-04-12T07:34:59.093530 #1] DEBUG -- Kontena::ServicePods::Creator: container created: test-fail-1
I, [2017-04-12T07:34:59.365626 #1]  INFO -- Kontena::ServicePods::Creator: service started: test-fail-1
D, [2017-04-12T07:34:59.366491 #1] DEBUG -- Kontena::Workers::ServicePodWorker: test-fail-1 mark as un-changed after ensure without error
D, [2017-04-12T07:34:59.385622 #1] DEBUG -- Kontena::Workers::ServicePodWorker: test-fail-1 mark as changed on event: create

D, [2017-04-12T07:35:10.759087 #1] DEBUG -- Kontena::Workers::ServicePodWorker: test-fail-1 update with changes -> apply
D, [2017-04-12T07:35:10.759671 #1] DEBUG -- Kontena::Workers::ServicePodWorker: state of test-fail-1: running
I, [2017-04-12T07:35:10.804600 #1]  INFO -- Kontena::Workers::ServicePodWorker: starting test-fail-1
I, [2017-04-12T07:35:10.856632 #1]  INFO -- Kontena::ServicePods::Starter: starting service: /test-fail-1
D, [2017-04-12T07:35:11.071737 #1] DEBUG -- Kontena::Workers::ServicePodWorker: test-fail-1 mark as un-changed after ensure without error

D, [2017-04-12T07:35:41.083002 #1] DEBUG -- Kontena::Workers::ServicePodWorker: test-fail-1 update without changes
```

And the `test-fail-1` container is and remains stopped, the worker will not try and re-start it anymore.